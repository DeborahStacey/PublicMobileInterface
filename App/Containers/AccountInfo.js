import React from 'react'
import { ScrollView, Text, Image, View } from 'react-native'
import { Images } from '../Themes'
import RoundedButton from '../Components/RoundedButton'
import {create} from 'apisauce'
import { Actions as NavigationActions } from 'react-native-router-flux'

//  Styles
import styles from './Styles/PresentationScreenStyle'

const db = create({
  baseURL: 'https://cat.ddns.net/Backend/api.php',
  headers: {'Content-Type': 'application/json'}
})

export default class AccountInfo extends React.Component {

  constructor (props) {
    super(props)
    this.state = {
      email: '',
      fname: '',
      lname: '',
      pnumber: '',
      street: '',
      city: '',
      unit: '',
      locID: '',
      postal: '',
      password: ''
    }
  }

  componentWillMount () {
    var that = this
    db.get('/user/view')
    .then(function (response) {
      var userData = response.data.userinfo
      that.setState({
        email: userData.email,
        fname: userData.firstname,
        lname: userData.lastname,
        pnumber: userData.phonenumber,
        street: userData.street,
        city: userData.city,
        unit: userData.unit,
        locID: parseInt(userData.locationid),
        postal: userData.postalcode
      })
    })
    .catch((error) => window.alert(error))
  }

  updatePhone (newPhone) {
    this.setState({
      pnumber: newPhone
    })
  }

  updateStreet (newStreet) {
    this.setState({
      street: newStreet
    })
  }

  updateUnit (newUnit) {
    this.setState({
      unit: newUnit
    })
  }

  updateCity (newCity) {
    this.setState({
      city: newCity
    })
  }

  updatePostal (newPostal) {
    this.setState({
      postal: newPostal
    })
  }

  updatePassword (pass) {
    this.setState({
      password: pass
    })
  }

  updateAccount (password) {
    var postObj = {
      'password': password,
      'phoneNumber': this.state.pnumber,
      'address': {
        'street': this.state.street,
        'unit': this.state.unit,
        'city': this.state.city,
        'postalCode': this.state.postal,
        'locationID': this.state.locID
      }
    }

    db.put('/user/update', postObj)
      .then((response) => window.alert(JSON.stringify(response.data)))
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          <View style={styles.centered}>
            <Image source={Images.accountInfoLogo} style={styles.profileImage} />
          </View>
          <Text style={styles.infoTitleText}>First Name</Text>
          <Text style={styles.infoText}>{this.state.fname}</Text>
          <Text style={styles.infoTitleText}>Last Name</Text>
          <Text style={styles.infoText}>{this.state.lname}</Text>
          <Text style={styles.infoTitleText}>Phone</Text>
          <Text style={styles.infoText}>{this.state.pnumber}</Text>
          <Text style={styles.infoTitleText}>Address</Text>
          <Text style={styles.infoText}>{this.state.street}</Text>
          <Text style={styles.infoText}>{this.state.unit}</Text>
          <Text style={styles.infoText}>{this.state.city}</Text>
          <Text style={styles.infoTitleText}>Postal</Text>
          <Text style={styles.infoText}>{this.state.postal}</Text>

          <RoundedButton onPress={NavigationActions.updateAccountInfo}>
            Edit
          </RoundedButton>
        </ScrollView>
      </View>
    )
  }
}
