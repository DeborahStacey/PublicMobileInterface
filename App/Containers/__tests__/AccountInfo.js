import 'react-native'
import React from 'react'
import AccountInfo from '../AccountInfo'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <AccountInfo />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
