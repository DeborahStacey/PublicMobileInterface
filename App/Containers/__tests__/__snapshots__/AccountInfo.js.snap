exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }>
      <Image
        source={1}
        style={
          Object {
            "borderRadius": 63,
            "height": 126,
            "width": 126,
          }
        } />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
      First Name
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 18,
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
              Lob
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
      Last Name
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 18,
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
              Riddle
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
      Email
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 18,
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
              LobRiddle@Gmail.com
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
      Address
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 18,
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
              45-375 Lottle Dr
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
      Cat Owner?
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#2C457F",
          "fontSize": 18,
          "marginVertical": 5,
          "textAlign": "left",
        }
      }>
              Yes
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        VIEW CATS
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        VIEW CAT GROUPS
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        EDIT
      </Text>
    </View>
  </ScrollView>
</View>
`;
